static/js/server_events_dispatch.js

```diff
@@ -48,62 +49,48 @@ exports.dispatch_normal_event = function dispatch_normal_event(event) {
         break;
 
     case 'realm':
-        if (event.op === 'update' && event.property === 'name') {
-            page_params.realm_name = event.value;
-            notifications.redraw_title();
-        } else if (event.op === 'update' && event.property === 'description') {
-            page_params.realm_description = event.value;
-            settings_org.update_realm_description(event.value);
-        } else if (event.op === 'update' && event.property === 'invite_required') {
-            page_params.realm_invite_required = event.value;
-        } else if (event.op === 'update' && event.property === 'invite_by_admins_only') {
-            page_params.realm_invite_by_admins_only = event.value;
-        } else if (event.op === 'update' && event.property === 'inline_image_preview') {
-            page_params.realm_inline_image_preview = event.value;
-        } else if (event.op === 'update' && event.property === 'inline_url_embed_preview') {
-            page_params.realm_inline_url_embed_preview = event.value;
-        } else if (event.op === 'update' && event.property === 'create_stream_by_admins_only') {
-            page_params.realm_create_stream_by_admins_only = event.value;
-            if (!page_params.is_admin) {
-                page_params.can_create_streams = !page_params.realm_create_stream_by_admins_only;
+        var realm_settings = {
+            add_emoji_by_admins_only: settings_emoji.update_custom_emoji_ui,
+            allow_edit_history: noop,
+            create_stream_by_admins_only: noop,
+            default_language: settings_org.reset_realm_default_language,
+            description: settings_org.update_realm_description,
+            email_changes_disabled: settings_org.toggle_email_change_display,
+            inline_image_preview: noop,
+            inline_url_embed_preview: noop,
+            invite_by_admins_only: noop,
+            invite_required: noop,
+            mandatory_topics: noop,
+            message_retention_days: settings_org.update_message_retention_days,
+            name: notifications.redraw_title,
+            name_changes_disabled: settings_org.toggle_name_change_display,
+            notifications_stream_id: noop,
+            restricted_to_domain: noop,
+            waiting_period_threshold: noop,
+        };
+        if (event.op === 'update' && _.has(realm_settings, event.property)) {
+            page_params['realm_' + event.property] = event.value;
+            realm_settings[event.property]();
+            if (event.property === 'create_stream_by_admins_only') {
```

Comment:
yep, that's the one.  Oh, I see what happened; you fixed this in the second commit, that's what confused me :)